nucypher/utilities/gas_strategies.py

```diff
@@ -64,23 +63,43 @@ def datafeed_fallback_gas_price_strategy(web3: Web3, transaction_params: TxParam
 }
 
 
-def wrap_web3_gas_strategy(web3_gas_strategy: Callable):
+def wrap_web3_gas_strategy(speed: Optional[str] = None):
     """
     Enriches the web3 exceptions thrown by gas strategies
     """
-    @functools.wraps(web3_gas_strategy)
+    web3_gas_strategy = __RAW_WEB3_GAS_STRATEGIES[speed]
+
     def _wrapper(*args, **kwargs):
         try:
             return web3_gas_strategy(*args, **kwargs)
         except ValidationError as e:
             raise GasStrategyError("Calling the web3 gas strategy failed, probably due to an unsynced chain.") from e
+
+    _wrapper.name = speed
+
     return _wrapper
 
 
-WEB3_GAS_STRATEGIES = {speed: wrap_web3_gas_strategy(strategy) for speed, strategy in __RAW_WEB3_GAS_STRATEGIES.items()}
+WEB3_GAS_STRATEGIES = {speed: wrap_web3_gas_strategy(speed) for speed in __RAW_WEB3_GAS_STRATEGIES}
 
 EXPECTED_CONFIRMATION_TIME_IN_SECONDS = {
     'slow': int(datetime.timedelta(hours=1).total_seconds()),
     'medium': int(datetime.timedelta(minutes=5).total_seconds()),
     'fast': 60
 }
+
+
+#
+# Fixed-price gas strategy
+#
+
+
+def construct_fixed_price_gas_strategy(gas_price, denomination: str = "wei") -> Callable:
+    gas_price_in_wei = Web3.toWei(gas_price, denomination)
```

Comment:
Well done - perhaps a variant can be built on top of this using values calculated in gas estimation metrics.

Reply:
What do you envision here? I don't understand

Reply:
Not to repeat myself here but:

> using values calculated in gas estimation metrics


Something like `construct_estimated_price_gas_strategy`

Reply:
> What do you envision here? 

A set of known defaults instead of a parameter for `gas_price`.

Reply:
I'm still not following you. What's the difference with, say, web3 gas strategies? Or are you saying something like `gas_strategy_50_gwei`?

Reply:
Hmm - yes...? If it does not make sense perhaps not a well formed idea after all.